tests/director/tests/formation_api_test.go

```diff
@@ -695,6 +698,134 @@ func TestSubaccountInAtMostOneFormationOfType(t *testing.T) {
 	require.Equal(t, 0, len(asaPage.Data))
 }
 
+func TestSystemInAtMostOneFormationOfType(t *testing.T) {
+	ctx := context.Background()
+	const (
+		firstFormationName  = "FIRST"
+		secondFormationName = "SECOND"
+	)
+
+	applicationTypes := []string{string(util.ApplicationTypeC4C), exceptionSystemType}
+	tenantId := tenant.TestTenants.GetDefaultTenantID()
+
+	formationTemplateName := "e2e-tests-formation-template-name"
+	formationTemplateInput := fixtures.FixFormationTemplateInputWithApplicationTypes(formationTemplateName, applicationTypes)
+
+	formationTemplateInputGQLString, err := testctx.Tc.Graphqlizer.FormationTemplateInputToGQL(formationTemplateInput)
+	require.NoError(t, err)
+
+	createFormationTemplateRequest := fixtures.FixCreateFormationTemplateRequest(formationTemplateInputGQLString)
+	formationTemplate := graphql.FormationTemplate{}
+
+	t.Logf("Create formation template with name: %q", formationTemplateName)
+	err = testctx.Tc.RunOperationWithoutTenant(ctx, certSecuredGraphQLClient, createFormationTemplateRequest, &formationTemplate)
+	defer fixtures.CleanupFormationTemplate(t, ctx, certSecuredGraphQLClient, &formationTemplate)
+	require.NoError(t, err)
+	require.NotEmpty(t, formationTemplate.ID)
+	require.NotEmpty(t, formationTemplate.Name)
+
+	in := graphql.FormationConstraintInput{
+		Name:            "TestSystemInAtMostOneFormationOfType",
+		ConstraintType:  graphql.ConstraintTypePre,
+		TargetOperation: graphql.TargetOperationAssignFormation,
+		Operator:        "IsNotAssignedToAnyFormationOfType",
```

Comment:
We should use the constant `IsNotAssignedToAnyFormationOfTypeOperator` for the operator name